app/modules/seed_library/endpoints_seed_library.py

```diff
@@ -0,0 +1,506 @@
+    db: AsyncSession = Depends(get_db),
+):
+    """
+    Update a Plant
+    **This endpoint is only usable by seed_library**
+    """
+    if not (
+        is_user_member_of_any_group(
+            user=user,
+            allowed_groups=[GroupType.seed_library],
+        )
+    ):
+        raise HTTPException(
+            status_code=403,
+            detail=f"You are not allowed to update plant {plant_id}",
+        )
+
+    try:
+        await cruds_seed_library.update_plant(
+            plant_id,
+            plant_edit,
+            db,
+        )
+    except ValueError as error:
+        raise HTTPException(status_code=400, detail=str(error))
+
+
+@module.router.patch(
+    "/seed_library/plants/{plant_id}/borrow",
+    status_code=204,
+)
+async def borrow_plant(
+    plant_id: uuid.UUID,
+    user: models_core.CoreUser = Depends(is_user()),
+    db: AsyncSession = Depends(get_db),
+):
+    """
+    Plant borrowed by the user (modify borrowing date, borrower and state)
+    """
+
+    try:
+        await cruds_seed_library.borrow_plant(user.id, plant_id, db)
+    except ValueError as error:
+        raise HTTPException(status_code=400, detail=str(error))
+
+
+@module.router.delete(
+    "/seed_library/plants/{plant_id}",
+    status_code=204,
+)
+async def delete_plant(
+    plant_id: uuid.UUID,
+    db: AsyncSession = Depends(get_db),
+    user: models_core.CoreUser = Depends(is_user()),
+):
+    """
+    Delete a Plant
+    **This endpoint is only usable by seed_library**
+    """
+
+    if not is_user_member_of_any_group(
+        user=user,
+        allowed_groups=[GroupType.seed_library],
+    ):
+        raise HTTPException(
+            status_code=403,
+            detail=f"You are not allowed to delete plant {plant_id}",
+        )
+
+    plant = await cruds_seed_library.get_plant_by_id(plant_id, db)
+    if plant is None:
+        raise HTTPException(404, "Plant does not exist.")
+
+    return await cruds_seed_library.delete_plant(
+        plant_id=plant_id,
+        db=db,
+    )
+
+
+# ---------------------------------------------------------------------------- #
+#                              SeedLibraryInformation                          #
+# ---------------------------------------------------------------------------- #
+
+
+@module.router.get(
+    "/seed_library/seed_library_information/",
+    response_model=schemas_seed_library.SeedLibraryInformation,
+    status_code=200,
+)
+async def get_seed_library_information(db: AsyncSession = Depends(get_db)):
+    try:
+        info = await tools.get_core_data(
+            schemas_seed_library.SeedLibraryInformation,
+            db,
+        )
+    except CoreDataNotFoundError:
+        await tools.set_core_data(
+            schemas_seed_library.SeedLibraryInformation(
+                facebook_url="",
+                forum_url="",
+                description="",
+                contact="",
+            ),
+            db,
+        )
```

Comment:
You don't need to check if the coredata object exist and initialize it with default values yourself. See [core_data.py#L35](https://github.com/aeecleclair/Hyperion/blob/cf3016c2ce3300462c5fcc7711fa3ce6a5348e37/app/types/core_data.py#L35)